### PR TITLE
[process-agent] Add telemetry to extractor and grpc server

### PR DESCRIPTION
**Pull request by just-chillin**
### What does this PR do?

This PR adds these metrics to the GRPCServer and the Extractor:

**Extractor:**
- `cache_size` - Gague for the cache size in the extractor
- `diff_dropped` - Count for how many times the extractor consumes the oldest diff in the `diffChan`
- `diff_chan_full` - Count for how many times the extractor tries to write to the `diffChan` but cannot because it is full. This should never happen!!!

**GRPCServer:**
- `invalid_version_errors` - Count for how many times an unexpected diff version is received, and subsequently the stream is dropped.
- `stream_send_errors` - Count for how many times the GRPCServer tried to send a message, but received an error.

### Motivation

Improved monitoring for processes-in-workloadmeta

### Additional Notes

Once this PR is merged, there may need to be changes to the process agent's deployment in order to get prometheus to recognize it's telemetry endpoint. Once this is done we _should_ be able to see these metrics in datadog 😃 

### Describe how to test/QA your changes

The goal here is to make sure that our metrics are being registered.

With the following yaml:
```yaml
telemetry: {enabled: true}
process_config:
  process_collection: {enabled: true}
  language_detection: {enabled: true} 
```
run the process agent. 

Then, on the same host run `curl localhost:6062/telemetry`, and make sure the above metrics are registered. The response should include the text below:
```
# HELP WorkloadMetaExtractor__cache_size The cache size for the workloadMetaExtractor
WorkloadMetaExtractor__cache_size 40
# HELP WorkloadMetaExtractor__diffchan_full The number of times the extractor was unable to write to the diffChan due to it being full. This should never happen.
# TYPE WorkloadMetaExtractor__diffchan_full counter
WorkloadMetaExtractor__diffchan_full 0
# HELP WorkloadMetaExtractor__invalid_version_errors The number of times the grpc server receives an entity diff that is from the future.
# TYPE WorkloadMetaExtractor__invalid_version_errors counter
WorkloadMetaExtractor__invalid_version_errors 0
# HELP WorkloadMetaExtractor__old_diff_dropped The number of times a diff is removed from the queue due to the diffChan being full.
# TYPE WorkloadMetaExtractor__old_diff_dropped counter
WorkloadMetaExtractor__old_diff_dropped 0
# HELP WorkloadMetaExtractor__stream_send_errors The number of times the grpc server has failed to send an entity diff to th--:-- --:--:-- --:--:e core agent.
# TYPE WorkloadMetaExtractor__stream_send_errors counter
WorkloadMetaExtractor__stream_send_errors 0
```

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/process/metadata/workloadmeta/extractor.go [hunk 1]
```diff
@@ -52,12 +59,17 @@ func NewWorkloadMetaExtractor(config config.ConfigReader) *WorkloadMetaExtractor
 		cache:        make(map[string]*ProcessEntity),
 		cacheVersion: 0,
 		// Keep only the latest diff in memory in case there's no consumer for it
-		diffChan: make(chan *ProcessCacheDiff, 1),
+		diffChan:        make(chan *ProcessCacheDiff, 1),
+		cacheSizeGuage:  telemetry.NewGauge(subsystem, "cache_size", nil, "The cache size for the workloadMetaExtractor"),
+		oldDiffDropped:  telemetry.NewSimpleCounter(subsystem, "old_diff_dropped", "The number of times a diff is removed from the queue due to the diffChan being full."),
```
Comment (xornivore) [inline pkg/process/metadata/workloadmeta/extractor.go:64]: Recognizing these are all mostly internal, but maybe just `diff_dropped` and `diff_chan_full`?

File: pkg/process/metadata/workloadmeta/grpc.go [hunk 2]
```diff
@@ -27,14 +28,19 @@ type GRPCServer struct {
...
 // NewGRPCServer creates a new instance of a GRPCServer
 func NewGRPCServer(config config.ConfigReader, extractor *WorkloadMetaExtractor) *GRPCServer {
 	l := &GRPCServer{
-		config:    config,
-		extractor: extractor,
-		server:    grpc.NewServer(),
+		config:              config,
+		extractor:           extractor,
+		server:              grpc.NewServer(),
+		invalidVersionError: telemetry.NewSimpleCounter(subsystem, "invalid_version_errors", "The number of times the grpc server receives an entity diff that is from the future."),
```
Comment (xornivore) [inline pkg/process/metadata/workloadmeta/grpc.go:42]:
```suggestion
		invalidVersionError: telemetry.NewSimpleCounter(subsystem, "invalid_version_errors", "The number of times the grpc server receives an entity diff that has an invalid version."),
```

**Review by xornivore — APPROVED** [review]
Looks great, some small naming/description nits.

File: pkg/process/metadata/workloadmeta/extractor.go (hunk 1 above)
Comment (mbotarro) [reply pkg/process/metadata/workloadmeta/extractor.go:64]: In both cases there's a message being dropped, so maybe `old_diff_dropped` and `new_diff_dropped`? It's expected for old diffs to be dropped if there's no consumers but a diff recently produced should never be dropped.

**Review by mbotarro — APPROVED** [review]
Agree with @xornivore's comments. Once they're addressed it should be good to be merged!

File: pkg/process/metadata/workloadmeta/extractor.go (hunk 1 above)
Comment (just-chillin) [reply pkg/process/metadata/workloadmeta/extractor.go:64]: Tbh, @xornivore's names feel a bit clearer to me. I think in the future the difference between an old and a new diff won't be super clear, and the help text explains clearly that it is dropped. I'll go ahead and change it to those.

**Comment by just-chillin** [comment]
Had to add a `t.Cleanup(telemetry.Reset)` because `NewExtractor` registers a telemetry handler

**Pull request merged**